test/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/forms/BusinessStartDateFormSpec.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.forms
+
+import org.scalatest.Matchers._
+import org.scalatestplus.play.PlaySpec
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.data.{Form, FormError}
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.forms.submapping.DateMapping
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.forms.validation.testutils.DataMap.DataMap
+import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.models.{BusinessStartDate, DateModel}
+
+
+class BusinessStartDateFormSpec extends PlaySpec with GuiceOneAppPerSuite {
+
+  import DateMapping._
+  import uk.gov.hmrc.incometaxsubscriptionselfemployedfrontend.forms.BusinessStartDateForm._
+
+
+  def form: Form[BusinessStartDate] ={
+    businessStartDateForm(BusinessStartDateForm.minStartDate.toString)
+  }
+
+  "The BusinessStartDateForm" should {
+    "transform a valid request to the date form case class" in {
+      val testDateDay = "31"
+      val testDateMonth = "05"
+      val testDateYear = "2017"
+
+      val testDateDay2 = "01"
+      val testDateMonth2 = "06"
+      val testDateYear2 = "2018"
+
+      val testInput = Map(
+        s"$startDate.$dateDay" -> testDateDay, s"$startDate.$dateMonth" -> testDateMonth, s"$startDate.$dateYear" -> testDateYear
+      )
+
+      val expected = BusinessStartDate(
+        DateModel(testDateDay, testDateMonth, testDateYear)
+      )
+
+      val actual = form.bind(testInput).value
+
+      actual shouldBe Some(expected)
+    }
+
+    "when testing the validation" should {
+
+      "output the appropriate error messages for the start date" when {
+        val empty = "error.date.empty"
+        val invalid = "error.date.empty"
+        val beforeMin = "error.business_accounting_period.minStartDate"
+
+        "the date is not supplied to the map" in {
+          form.bind(DataMap.EmptyMap).errors must contain(FormError(startDate, empty))
+        }
+
+        "the date supplied is empty" in {
+          form.bind(DataMap.emptyDate(startDate)).errors must contain(FormError(startDate, empty))
+        }
+
+        "it is an invalid date" in {
+          val invalidTest = form.bind(DataMap.date(startDate)("29", "2", "2017"))
+          invalidTest.errors must contain(FormError(startDate, invalid))
+        }
+
+        "if it is not before the " in {
```

Comment:
Fix test name

Reply:
Why is this test here? Shouldn't this test be testing the error message if after a certain date?